### PR TITLE
200 add amount of students to course info page

### PR DESCRIPTION
**Pull request by essal0815**
#200 
The cards of each course got additional info in the left corner, which shows a member counter.

screen shot 2018-06-20 at 12 11 10

On hovering it shows the first 10 members in a tooltip
screen shot 2018-06-20 at 12 11 19

On clicking it opens a modal with all members / students listed in 3 columns
screen shot 2018-06-20 at 12 10 58

If no students are in the course yet, the person with appropriate rights get the option to edit the course directly.
screen shot 2018-06-20 at 12 11 29

### CODE REVIEW

File: static/styles/courses/course.scss [hunk 1]
```diff
@@ -156,6 +156,23 @@
     }
 }
 
+#member-list {
+    columns: 3;
```
Comment (schurigh) [inline static/styles/courses/course.scss]: numbers wont work in chrome/opera on column 2 and 3

File: controllers/courses.js [hunk 2]
```diff
@@ -162,7 +162,10 @@ router.get('/', function(req, res, next) {
                 $or: [
                     { userIds: res.locals.currentUser._id },
                     { teacherIds: res.locals.currentUser._id }
-               ]
+                ],
+                $populate: [
```
Comment (schurigh) [inline controllers/courses.js]: Loading 16 courses with 25 students each doesn't really effect loading time/performance, <100ms at max, so should be okay

File: views/lib/sc-card_additionalInfo/memberCount.hbs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                          <li>{{this.displayName}}</li>
+                        {{else}}
+                          <li>{{this.firstName}} {{this.lastName}}</li>
+                         {{/if}}
+                        {{/each}}
+                        </ol>
+                    {{else}}
+                        Noch keine Teilnehmer
+                    {{/if}}"
+       data-title="{{#each (truncateArray this.members 10)}}
```
Comment (schurigh) [inline views/lib/sc-card_additionalInfo/memberCount.hbs]: Remove hover/tooltip because it's useless?

File: views/courses/overview.hbs [hunk 4]
```diff
@@ -1,8 +1,16 @@
...
+    {{/content}}
+    {{#content "styles" mode="append"}}
+        <link rel="stylesheet" href="/styles/courses/course.css"/>
+    {{/content}}
+
     {{#content "page"}}
         <section class="section-courses">
                 {{#if courses}}
-                    {{> 'dashboard/components/card_section' section-id="courses" title="" content=courses link-text="Öffnen"}}
+                    {{> 'dashboard/components/card_section' section-id="courses" title="" content=courses link-text="Öffnen" additionalInfoName="memberCount"}}
```
Comment (schurigh) [inline views/courses/overview.hbs]: add member info to substitution courses as well

File: views/lib/sc-card_additionalInfo/memberCount.hbs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{{#if this.memberAmount includeZero=true}}
+    <a href="" class="btn-member" data-id="{{this._id}}">
```
Comment (schurigh) [inline views/lib/sc-card_additionalInfo/memberCount.hbs]: remove onclick/link if 0 members? discussable

**Comment by Langleu** [comment]
why don't you make an extra route just for getting the students? I looked shortly through the scripts and there are two other scripts referencing the ..../json route and they don't need the populated userIds Array.

image
"Kurs bearbeiten"? I don't really understand the flow here... the teacher wants to look who is in his course and ends up having the option to edit the course? 
Maybe add an icon like at news instead of having that option in this modal.

You introduced two helpers, which you don't use yourself, do we need them?

Please call the modal as followed: (iOS had some problems with the other way)
$memberModal.appendTo('body').modal('show');

**Pull request merged**